library/pkcs11_client.c

```diff
@@ -265,12 +265,88 @@ static int pkcs11_sign( void *ctx_arg,
     return( pkcs11_err_to_mbedtls_pk_err( rv ) );
 }
 
+static int pkcs11_verify( void *ctx_arg,
+                        mbedtls_md_type_t md_alg,
+                        const unsigned char *hash, size_t hash_len,
+                        const unsigned char *sig, size_t sig_len)
+{
+    mbedtls_pk_pkcs11_context_t *ctx = ctx_arg;
+    CK_RV rv;
+    CK_MECHANISM mechanism = {0, NULL_PTR, 0};
+    unsigned char *decoded_sig = NULL_PTR;
```

Comment:
Just a style matter, but `NULL_PTR` is a Cryptoki thing, so this should either be `CK_BYTEPTR decoded_sig = NULL_PTR` or `unsigned char *decoded_sig = NULL`.